web-api/benches/stateless_personalization.rs

```diff
@@ -65,33 +65,33 @@ criterion_group!(
     bench_derive_interests_with_identical_embedding_128_40,
 );
 
-fn derive_interests_with_orthogonal_embeddings(embedding_size: usize, interest_size: usize) {
-    let system = CoiConfig::default().build();
-    let timestamp = Utc::now();
-    let history = (0..interest_size)
-        .map(|i| {
-            let mut embedding = vec![0.0; embedding_size];
-            embedding[i] = 1.0;
-            let embedding = Embedding1::from(embedding).normalize().unwrap();
-            (timestamp, embedding)
-        })
-        .collect_vec();
-    bench_derive_interests(black_box(&system), black_box(history));
-}
-
 macro_rules! bench_orthogonal {
     ($($function: ident, $embedding_size: expr, $interest_size: expr);+ $(;)?) => {$(
         fn $function(c: &mut Criterion) {
+            let system = CoiConfig::default().build();
+            let timestamp = Utc::now();
+            let history = (0..$interest_size)
+                .map(|i| {
+                    let mut embedding = vec![0.0; $embedding_size];
+                    embedding[i] = 1.0;
+                    let embedding = Embedding1::from(embedding).normalize().unwrap();
+                    (timestamp, embedding)
+                })
+                .collect_vec();
+
             let name = format!(
                 "derive {} interests with orthogonal embeddings ({})",
                 $interest_size,
                 $embedding_size,
             );
-            c.bench_function(&name, |b| {
-                b.iter(|| {
-                    derive_interests_with_orthogonal_embeddings($embedding_size, $interest_size)
-                })
-            });
+            c.bench_function(
+                &name,
+                |b| b.iter_batched(
+                    || history.clone(),
+                    |history| bench_derive_interests(black_box(&system), black_box(history)),
```

Comment:
:art:

there is no reason to black_box the history , or any "batch" data passed with `iter_batched` 

```suggestion
                    |history| bench_derive_interests(black_box(&system), history),
```

Reply:
ah i see, i didn't know that `iter_batched()` already wraps the input in a `black_box()`. i still like to keep it since criterion uses its own `black_box()` which they say isn't perfect, and we use the compiler built-in `black_box()` from `std`.